dom.go

```diff
@@ -94,7 +105,9 @@ type Keyer interface {
 //
 // If the underlying value is not one of these types, the code handling the
 // value is expected to panic.
```

Comment:
It's impossible for the underlying value to not be one of these types now, so we should replace this comment with one that says the compiler enforces only the above types can implement this interface.